lib/ansible/modules/cloud/vmware/vmware_host_hyperthreading.py

```diff
@@ -0,0 +1,246 @@
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    esxi_hostname: '{{ esxi_hostname }}'
+    enabled: false
+    validate_certs: no
+  delegate_to: localhost
+
+- name: Disable Hyperthreading for all host systems from cluster
+  vmware_host_hyperthreading:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    cluster_name: '{{ cluster_name }}'
+    enabled: false
+    validate_certs: no
+  delegate_to: localhost
+'''
+
+RETURN = r'''
+result:
+    description: metadata about host system's Hyperthreading configuration
+    returned: always
+    type: dict
+    sample: {
+        "esxi01": {
+            "msg": "Hyperthreading is already enabled and active for host 'esxi01'",
+            "current_state": "enabled",
+            "previous_state": "enabled",
+            "desired_state": "enabled",
+            "error": "",
+        },
+    }
+'''
+
+try:
+    from pyVmomi import vmodl
+except ImportError:
+    pass
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vmware import PyVmomi, vmware_argument_spec, wait_for_task, TaskError
+from ansible.module_utils._text import to_native
+
+
+class VmwareHostHyperthreading(PyVmomi):
+    """
+    Class to manage Hyperthreading for an ESXi host system
+    """
+    def __init__(self, module):
+        super(VmwareHostHyperthreading, self).__init__(module)
+        cluster_name = self.params.get('cluster_name')
+        esxi_host_name = self.params.get('esxi_hostname')
+        self.hosts = self.get_all_host_objs(cluster_name=cluster_name, esxi_host_name=esxi_host_name)
+        if not self.hosts:
+            self.module.fail_json(msg="Failed to find host system with given configuration.")
+
+    def ensure(self):
+        """
+        Function to manage Hyperthreading for an ESXi host system
+        """
+        results = dict(changed=False, result=dict())
+        enable_hyperthreading = self.params.get('enabled')
+        host_change_list = []
+        for host in self.hosts:
+            changed = False
+            results['result'][host.name] = dict(msg='', error='')
+
+            hyperthreading_info = host.config.hyperThread
+
+            if enable_hyperthreading:
+                results['result'][host.name]['desired_state'] = "enabled"
+                # Don't do anything if Hyperthreading is already enabled
+                if hyperthreading_info.config and hyperthreading_info.active:
+                    results['result'][host.name]['previous_state'] = "enabled"
+                    results['result'][host.name]['current_state'] = "enabled"
+                    results['result'][host.name]['msg'] = "Hyperthreading is already enabled and active for host '%s'" % \
+                                                          host.name
+                elif hyperthreading_info.config and hyperthreading_info.active is False:
```

Comment:
isn't it better practice to do `and not hyperthreading_info.active` instead?

Reply:
Yes, makes sense. I've changed it. Thanks.